tf2_bullet/bullet-extras.cmake

```diff
@@ -26,9 +26,14 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+# Bullet CMake only exports old-style CMake variables.
+# This script makes an imported target for them.
+
 if(WIN32)
   set(BULLET_ROOT $ENV{ChocolateyInstall}/lib/bullet)
 endif()
 find_package(Bullet REQUIRED)
 
-include_directories(SYSTEM ${BULLET_INCLUDE_DIRS})
+add_library(tf2_bullet::Bullet INTERFACE IMPORTED)
+target_include_directories(tf2_bullet::Bullet INTERFACE ${BULLET_INCLUDE_DIRS})
+target_link_libraries(tf2_bullet::Bullet INTERFACE ${BULLET_LIBRARIES})
```

Comment:
Windows CI is failing, and I suspect it's because I added this line. Previously tf2_geometry_msgs wasn't having downstream targets link against bullet, which I suspect is incorrect.

The `BulletConfig.cmake` file provided by https://github.com/ros2/choco-packages/releases/download/2020-02-24/bullet.2.89.0.nupkg seems odd, the `BULLET_LIBRARIES` variable includes both static libraries (`.lib`) and dynamic (`.dll`), but of the same libraries. It looks like it forces linking against both the static and the dynamic versions of the same libraries

```
BulletSoftBody.lib(btSoftBodyHelpers.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in test_tf2_bullet.obj [C:\ci\ws\build\tf2_bullet\test_bullet.vcxproj]

BulletSoftBody.lib(btSoftBody.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in test_tf2_bullet.obj [C:\ci\ws\build\tf2_bullet\test_bullet.vcxproj]

msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::basic_streambuf<char,struct std::char_traits<char> >(void)" (??0?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAA@XZ) already defined in BulletSoftBody.lib(btSoftBodyHelpers.obj) [C:\ci\ws\build\tf2_bullet\test_bullet.vcxproj]

msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: virtual __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::~basic_streambuf<char,struct std::char_traits<char> >(void)" (??1?$basic_streambuf@DU?$char_traits@D@std@@@std@@UEAA@XZ) already defined in BulletSoftBody.lib(btSoftBodyHelpers.obj) [C:\ci\ws\build\tf2_bullet\test_bullet.vcxproj]

msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: virtual __cdecl std::basic_ios<char,struct std::char_traits<char> >::~basic_ios<char,struct std::char_traits<char> >(void)" (??1?$basic_ios@DU?$char_traits@D@std@@@std@@UEAA@XZ) already defined in BulletSoftBody.lib(btSoftBodyHelpers.obj) [C:\ci\ws\build\tf2_bullet\test_bullet.vcxproj]

msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: void __cdecl std::basic_ostream<char,struct std::char_traits<char> >::_Osfx(void)" (?_Osfx@?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAXXZ) already defined in BulletSoftBody.lib(btSoftBodyHelpers.obj) [C:\ci\ws\build\tf2_bullet\test_bullet.vcxproj]

msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __cdecl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(int)" (??6?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAAEAV01@H@Z) already defined in BulletSoftBody.lib(btSoftBodyHelpers.obj) [C:\ci\ws\build\tf2_bullet\test_bullet.vcxproj]
```


```cmake
get_filename_component(_PREFIX "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)

message("LOL!!")

set (BULLET_INCLUDE_DIRS "${_PREFIX}/include")

set(BULLET_DYNAMICS_LIBRARY
  debug
    "${_PREFIX}/lib/Bullet3Dynamics_Debug.dll"
    "${_PREFIX}/lib/Bullet3Dynamics_Debug.lib"
    "${_PREFIX}/lib/BulletDynamics_Debug.dll"
    "${_PREFIX}/lib/BulletDynamics_Debug.lib"
    "${_PREFIX}/lib/BulletInverseDynamics_Debug.dll"
    "${_PREFIX}/lib/BulletInverseDynamics_Debug.lib"
  optimized
    "${_PREFIX}/lib/Bullet3Dynamics.dll"
    "${_PREFIX}/lib/BulletDynamics.dll"
    "${_PREFIX}/lib/BulletDynamics.lib"
    "${_PREFIX}/lib/BulletInverseDynamics.dll"
    "${_PREFIX}/lib/BulletInverseDynamics.lib"
)

set(BULLET_COLLISION_LIBRARY
  debug
    "${_PREFIX}/lib/Bullet3Collision_Debug.dll"
    "${_PREFIX}/lib/Bullet3Collision_Debug.lib"
    "${_PREFIX}/lib/BulletCollision_Debug.dll"
    "${_PREFIX}/lib/BulletCollision_Debug.lib"
  optimized
    "${_PREFIX}/lib/Bullet3Collision.dll"
    "${_PREFIX}/lib/Bullet3Collision.lib"
    "${_PREFIX}/lib/BulletCollision.dll"
    "${_PREFIX}/lib/BulletCollision.lib"
)

set(BULLET_MATH_LIBRARY
  debug
    "${_PREFIX}/lib/LinearMath_Debug.dll"
    "${_PREFIX}/lib/LinearMath_Debug.lib"
  optimized
    "${_PREFIX}/lib/LinearMath.dll"
    "${_PREFIX}/lib/LinearMath.lib"
)

set(BULLET_SOFTBODY_LIBRARY
  debug
    "${_PREFIX}/lib/BulletSoftBody_Debug.dll"
    "${_PREFIX}/lib/BulletSoftBody_Debug.lib"
  optimized
    "${_PREFIX}/lib/BulletSoftBody.dll"
    "${_PREFIX}/lib/BulletSoftBody.lib"
)

set (BULLET_LIBRARIES
  debug
    "${_PREFIX}/lib/Bullet2FileLoader_Debug.dll"
    "${_PREFIX}/lib/Bullet2FileLoader_Debug.lib"

    "${_PREFIX}/lib/Bullet3Common_Debug.dll"
    "${_PREFIX}/lib/Bullet3Common_Debug.lib"
    "${_PREFIX}/lib/Bullet3Geometry_Debug.dll"
    "${_PREFIX}/lib/Bullet3Geometry_Debug.lib"
    "${_PREFIX}/lib/Bullet3OpenCL_clew_Debug.dll"
    "${_PREFIX}/lib/Bullet3OpenCL_clew_Debug.lib"
  optimized
    "${_PREFIX}/lib/Bullet2FileLoader.dll"
    "${_PREFIX}/lib/Bullet2FileLoader.lib"
    "${_PREFIX}/lib/Bullet3Common.dll"
    "${_PREFIX}/lib/Bullet3Common.lib"
    "${_PREFIX}/lib/Bullet3Geometry.dll"
    "${_PREFIX}/lib/Bullet3Geometry.lib"
    "${_PREFIX}/lib/Bullet3OpenCL_clew.dll"
    "${_PREFIX}/lib/Bullet3OpenCL_clew.lib"
)

set (BULLET_INCLUDE_DIR "${BULLET_INCLUDE_DIRS}")
set (BULLET_INCLUDE_DIRS "${BULLET_INCLUDE_DIRS}")
set (BULLET_INCLUDE_DIR "${BULLET_INCLUDE_DIRS}")
set (BULLET_LIBRARY "${BULLET_LIBRARIES}")
set (BULLET_LIBRARIES "${BULLET_LIBRARIES}")
set (BULLET_LIBRARY "${BULLET_LIBRARIES}")
```

Reply:
The file comes from here: looks like it's something we made https://github.com/ros2/choco-packages/blob/3c6b2741cfec72487257d6a064e886cb6245d2b3/package/bullet/share/cmake/BulletConfig.cmake

Reply:
https://github.com/ros2/choco-packages/pull/18